ts/packages/anchor/src/program/accounts-resolver.ts

```diff
@@ -232,31 +230,17 @@ export class AccountsResolver<IDL extends Idl, I extends AllInstructions<IDL>> {
 export class AccountStore<IDL extends Idl> {
   private _cache = new Map<string, any>();
 
-  // todo: don't use the progrma use the account namespace.
-  constructor(
-    private _provider: Provider,
-    private _accounts: AccountNamespace<IDL>
-  ) {}
+  // TODO: don't use the program use the account namespace.
+  constructor(private _accounts: AccountNamespace<IDL>) {}
 
   public async fetchAccount<T = any>(
     name: string,
     publicKey: PublicKey
   ): Promise<T> {
     const address = publicKey.toString();
     if (!this._cache.has(address)) {
-      if (name === "TokenAccount") {
-        const accountInfo = await this._provider.connection.getAccountInfo(
-          publicKey
-        );
-        if (accountInfo === null) {
-          throw new Error(`invalid account info for ${address}`);
-        }
-        const data = coder().accounts.decode("token", accountInfo.data);
```

Comment:
Could just decode token accounts without any coder, not sure if it's needed though.

Reply:
if this is going to be removed, would be worth also investigating where and why `TokenAccounts` are being added to the account store cache and managing that too.

Reply:
maybe @armaniferrante has insight there

Reply:
Completely removing this caused a problem that I don't remember so I added a coder for only what we need and nothing more in https://github.com/coral-xyz/anchor/pull/2155/commits/57f0f7e660ce3a559fa6b225ad2c772de5a7db13